src/generate.c

```diff
@@ -339,5 +345,7 @@ int main(int argc, char** argv)
         netplan_parser_clear(&npp);
     if (np_state)
         netplan_state_clear(&np_state);
+    if (sriov_pfs)
+        g_hash_table_destroy(sriov_pfs);
```

Comment:
nitpick: the hashtable should be cleared before the state.

There's a brief window where we have a hashtable full of dangling pointers. This isn't a big deal here but is easily avoidable.